src/compiler/checker.ts

```diff
@@ -5667,6 +5752,191 @@ namespace ts {
             return links.resolvedType;
         }
 
+        function getTypeFromSpreadTypeLiteral(node: Node, aliasSymbol?: Symbol, aliasTypeArguments?: Type[]): Type {
+            let members: Map<Symbol>;
+            let stringIndexInfo: IndexInfo;
+            let numberIndexInfo: IndexInfo;
+            const spreads: Type[] = [];
+            for (const member of (node as TypeLiteralNode).members) {
+                if (member.kind === SyntaxKind.SpreadTypeElement) {
+                    if (members) {
+                        spreads.push(createAnonymousType(node.symbol, members, emptyArray, emptyArray, stringIndexInfo, numberIndexInfo));
+                        members = undefined;
+                        stringIndexInfo = undefined;
+                        numberIndexInfo = undefined;
+                    }
+                    spreads.push(getTypeFromTypeNode((member as SpreadTypeElement).type));
+                }
+                else if (member.kind === SyntaxKind.IndexSignature) {
+                    const index = member as IndexSignatureDeclaration;
+                    if (index.parameters.length === 1) {
+                        const parameter = index.parameters[0];
+                        if (parameter && parameter.type) {
+                            const indexInfo = createIndexInfo(index.type ? getTypeFromTypeNode(index.type) : anyType,
+                                                              (getModifierFlags(index) & ModifierFlags.Readonly) !== 0, index);
+                            if (parameter.type.kind === SyntaxKind.StringKeyword) {
+                                stringIndexInfo = indexInfo;
+                            }
+                            else {
+                                numberIndexInfo = indexInfo;
+                            }
+                        }
+                    }
+                }
+                else if (member.kind !== SyntaxKind.CallSignature && member.kind !== SyntaxKind.ConstructSignature) {

```

Comment:
In fact, index signatures are illegal too, so I don't need the code to create those either. I added the error and got rid of the index signature code.